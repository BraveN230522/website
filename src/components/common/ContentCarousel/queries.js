import { useQuery } from '@apollo/react-hooks';
import { get } from 'lodash';

import Loader from '@components/Loader';
import console from '@utils/console';
import { GET_NEWS } from '../../../graphql/queries/news';

const NewsQuery = ({ children, interestId, interestType, shouldRefetch, refetchInterestId, refetchInterestType }) => {
  const { loading, error, data, refetch } = useQuery(GET_NEWS, {
    variables: { interestId, interestType },
  });

  // TODO: deal with loading and error
  if (loading) {
    return <Loader />;
  }

  if (error) {
    console.error(error, 'error on league news');
    if (shouldRefetch && refetchInterestId > -1 && refetchInterestType) {
      refetch({
        interestId: refetchInterestId,
        interestType: refetchInterestType,
      });
    }
  }

  const queryData = get(data, 'getNews', []);

  if (queryData?.length === 0 && shouldRefetch && refetchInterestId > -1 && refetchInterestType) {
    refetch({
      interestId: refetchInterestId,
      interestType: refetchInterestType,
    });
  }

  return children(queryData);
};

export { NewsQuery };
