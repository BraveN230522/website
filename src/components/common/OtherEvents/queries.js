import gql from 'graphql-tag';
import { useQuery } from '@apollo/react-hooks';
import console from '@utils/console';

const GET_NEARBY_EVENTS_BY_CITY = gql`
  query nearbyEventsByCityId($cityId: Int!) {
    nearbyEventsByCityId(cityId: $cityId) {
      eventDateTime
      eventID
      eventImage
      eventName
      venue {
        venueAddress
        venueID
        venueImage
        venueName
      }
      fromPrice {
        amount
        currencyId
      }
    }
  }
`;

const GET_EVENTS_NEAR_HOTEL = gql`
  query eventsNearHotel($hotelId: String!) {
    eventsNearHotel(hotelId: $hotelId) {
      eventDateTime
      eventID
      eventImage
      eventName
      fromPrice {
        amount
        currencyId
      }
      venue {
        venueName
      }
    }
  }
`;

export const NearbyEventsByCityQuery = ({ children, cityId }) => {
  const { loading, error, data } = useQuery(GET_NEARBY_EVENTS_BY_CITY, {
    variables: { cityId },
  });

  // TODO: deal with loading and error
  if (loading) {
    console.info('Loading');
  }

  if (error) {
    console.error(error, 'error');
  }

  const nearbyCities = ((data && data.nearbyEventsByCityId) || []).sort(
    (a, b) => new Date(a.eventDateTime) - new Date(b.eventDateTime)
  );

  return children({
    nearbyCities,
  });
};

export const NearbyEventsByHotelQuery = ({ children, hotelId }) => {
  const { loading, error, data } = useQuery(GET_EVENTS_NEAR_HOTEL, {
    variables: { hotelId },
  });

  // TODO: deal with loading and error
  if (loading) {
    console.info('Loading');
  }

  if (error) {
    console.error(error, 'error');
  }

  const nearbyEvents = (data && data.eventsNearHotel) || [];

  return children({
    nearbyEvents,
  });
};

export default NearbyEventsByCityQuery;
