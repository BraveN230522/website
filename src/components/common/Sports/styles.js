import styled from '@emotion/styled';

const getCommonFontStyles = ({ theme }) => `
  color: ${theme.colors.white};
  font-family: ${theme.fonts.Inter};
`;

const SportsStyles = styled.section`
  background: ${({ theme }) => theme.colors.black};
  padding: 56px 0 30px 0;

  ${({ theme }) => theme.media.tablet`
    padding: 24px 0 22px;
  `}
`;

const SportsContainer = styled.div`
  ${({ theme }) => theme.setContainer()}

  padding-left: 15px;
  padding-right: 15px;
`;

const SportsHeader = styled.header`
  align-items: center;
  display: flex;
  flex: 1;

  ${({ theme }) =>
    theme.generateBreakpoint(
      1220,
      `
        margin: 0 10px;
      `
    )};
`;

const SportsImage = styled.img`
  max-width: 61px;
  width: 100%;
  margin-bottom: 22px;
`;

const SportsTitle = styled.span`
  ${getCommonFontStyles}
  font-size: 22px;
  font-weight: 400;
  letter-spacing: -0.1px;
  line-height: 29px;
  margin-bottom: 37px;
`;

const SportsArticle = styled.article`
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  margin: 30px 0 15px;
`;

const SportsContentItems = styled.a`
  align-items: center;
  display: flex;
  flex-direction: column;
  min-height: 150px;
  width: 12.5%;

  &:hover {
    svg > g,
    svg + span {
      opacity: 1;
      transition: all ease-in 0.3s;
    }
  }

  ${({ theme }) => theme.media.tablet`
    width: 33%;
    min-height: 121px;
    margin-bottom: 41px;
  `}
`;

const SportsDescription = styled.span`
  ${getCommonFontStyles}
  display: block;
  font-size: 13px;
  font-weight: 400;
  line-height: 17px;
  opacity: 0.7;
  text-align: center;
`;

export {
  SportsStyles,
  SportsContainer,
  SportsHeader,
  SportsImage,
  SportsTitle,
  SportsArticle,
  SportsContentItems,
  SportsDescription,
};
