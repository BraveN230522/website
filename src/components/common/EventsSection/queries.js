import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import Loader from '@components/Loader';

import { GET_EVENTS } from '@graphql/queries/events';
import { GET_TOP_CITIES } from '@graphql/queries/cities';

const EventsQuery = ({ children }) => {
  const { loading, error, data } = useQuery(GET_EVENTS);

  if (loading) {
    return <Loader />;
  }

  if (error) {
    const msg = `Error loading sports`;
    // here set global error state (with useContext or return a toast notification or other error component
    return <div>{msg}</div>;
  }

  if (!data.allExperiences) {
    data.allExperiences = [];
  }

  if (!data.allPopularEvents) {
    data.allPopularEvents = [];
  }

  // The data is not paginated on backend, it's keeped in FE application memory
  return children({
    ...data,
  });
};

const TopCitiesQuery = ({ children }) => {
  const { loading, error, data } = useQuery(GET_TOP_CITIES);

  if (loading) {
    // return <div>Loading...</div>;
    return null;
  }

  if (error) {
    const msg = `Error loading cities`;
    // here set global error state (with useContext or return a toast notification or other error component
    return <div>{msg}</div>;
  }

  if (!data.topCities) {
    data.topCities = [];
  }

  // The data is not paginated on backend, it's keeped in FE application memory
  return children(data.topCities);
};

export { EventsQuery, TopCitiesQuery };
