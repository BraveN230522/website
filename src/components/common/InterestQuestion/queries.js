import { useQuery } from '@apollo/react-hooks';
import { get } from 'lodash';
import { GET_INTEREST_QUESTIONS } from '../../../graphql/queries/interest-data';

const InterestQuestionsQuery = ({ children, interestId, interestType }) => {
  const { loading, error, data } = useQuery(GET_INTEREST_QUESTIONS, {
    variables: { interestId, interestType },
  });

  // TODO: deal with loading and error
  if (loading) {
    console.info('Loading');
  }

  if (error) {
    console.error(error, `Error fetching interest questions for interest ${interestType} on id ${interestId}`);
  }

  const queryData = get(data, 'interestQuestions', []);

  return children(queryData);
};

export default InterestQuestionsQuery;
