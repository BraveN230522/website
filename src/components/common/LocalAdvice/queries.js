import { useQuery } from '@apollo/react-hooks';
import { get } from 'lodash';
import console from '@utils/console';
import { GET_INTEREST_DATA } from '../../../graphql/queries/interest-data';

const InterestDataQuery = ({ children, interestId, interestType }) => {
  const { loading, error, data } = useQuery(GET_INTEREST_DATA, {
    variables: { interestId, interestType },
  });

  // TODO: deal with loading and error
  if (loading) {
    console.info('Loading');
  }

  if (error) {
    console.error(error, `Error fetching interest data for interest ${interestType} on id ${interestId}`);
  }

  const queryData = get(data, 'interestData', []);

  return children(queryData);
};

export default InterestDataQuery;
