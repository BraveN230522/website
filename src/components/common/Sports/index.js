import React, { useState, useEffect } from 'react';
import Link from 'next/link';
import { adopt } from 'react-adopt';

import More from '@components/common/More';
import { useMediaQuery } from '@utils/hooks/useMedia';

import { SportsQuery } from './queries';

import {
  SportsStyles,
  SportsContainer,
  SportsHeader,
  SportsTitle,
  SportsArticle,
  SportsContentItems,
  SportsDescription,
  SportsImage,
} from './styles';

const SportsData = adopt({
  sports: ({ render }) => <SportsQuery>{render}</SportsQuery>,
});

const Sports = () => {
  const isTablet = useMediaQuery('(max-width: 960px)');

  const getNumberOfItems = () => (isTablet ? 9 : 8);

  const [limit, setLimit] = useState(getNumberOfItems());

  useEffect(() => {
    setLimit(getNumberOfItems());
  }, [isTablet]);

  return (
    <SportsStyles>
      <SportsContainer>
        <SportsHeader>
          <SportsTitle>Get tickets by Sport</SportsTitle>
        </SportsHeader>

        <SportsData>
          {({ sports: { allSports } }) => {
            return (
              <>
                <SportsArticle>
                  {!!allSports.length &&
                    Array.from(allSports)
                      .slice(0, limit)
                      .map(({ sportID, sportName, sportIconURLV4 }) => (
                        <Link href="/" passHref key={sportID}>
                          <SportsContentItems aria-label="sport ticket">
                            <SportsImage src={sportIconURLV4} />
                            <SportsDescription>{sportName}</SportsDescription>
                          </SportsContentItems>
                        </Link>
                      ))}
                </SportsArticle>
              </>
            );
          }}
        </SportsData>
      </SportsContainer>
    </SportsStyles>
  );
};

export default Sports;
