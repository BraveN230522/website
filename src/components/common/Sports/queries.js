import { GET_SPORTS } from '@graphql/queries/sports';
import { useQuery } from '@apollo/react-hooks';

const SportsQuery = ({ children }) => {
  const { loading, error, data } = useQuery(GET_SPORTS);

  if (loading) {
    return null;
    // return <div>Loading...</div>;
  }

  if (error || !data.allSports) {
    const msg = `Error loading sports`;
    // here set global error state (with useContext or return a toast notification or other error component
    return <div>{msg}</div>;
  }

  // The data is not paginated on backend, it's keeped in FE application memory
  return children({
    ...data,
  });
};

export { SportsQuery };
