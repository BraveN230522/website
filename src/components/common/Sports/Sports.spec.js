import React from 'react';
import { render, fireEvent, waitFor, screen, act } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';

import withGlobals from '@swiamTestUtils/withGlobals';
import { GET_SPORTS } from '@graphql/queries/sports';

import Sports from './index';

const mocks = [
  {
    request: {
      query: GET_SPORTS,
      variables: {},
    },
    result: {
      data: {
        allSports: Array.from({ length: 20 }, (v, k) => ({
          sportID: k,
          sportName: `Sport ${k}`,
          sportIconURL: 'Sport club',
          sportIconURLV4: 'Sport club',
        })),
      },
    },
  },
];

describe('<Sports />', () => {
  it('Renders sports unchanged', async () => {
    let container;
    let getByText;

    act(() => {
      const component = render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Sports />
        </MockedProvider>,
        {
          wrapper: withGlobals,
        }
      );

      container = component.container;
      getByText = component.getByText;
    });

    expect(container).toHaveTextContent('Loading');

    await waitFor(() => getByText(/Sport 1/));

    expect(container).toMatchSnapshot();
  });

  it('Shows more tickets items when more option is clickedRenders sports unchanged', async () => {
    let container;
    let getByText;

    act(() => {
      const component = render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Sports />
        </MockedProvider>,
        {
          wrapper: withGlobals,
        }
      );

      container = component.container;
      getByText = component.getByText;
    });

    expect(container).toHaveTextContent('Loading');

    await waitFor(() => getByText(/Sport 1/));

    fireEvent.click(screen.getByLabelText('more sports'));

    await waitFor(() => screen.getAllByLabelText('sport ticket').length > 10);

    expect(container).toMatchSnapshot();
  });
});
