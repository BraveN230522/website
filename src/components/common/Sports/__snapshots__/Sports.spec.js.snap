// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sports /> Renders sports unchanged 1`] = `
.emotion-66 {
  background: #000000;
  padding: 56px 0 30px 0;
}

@media (max-width:60em) {
  .emotion-66 {
    padding: 24px 0 22px;
  }
}

.emotion-64 {
  max-width: 1255px;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  padding-left: 30px;
  padding-right: 30px;
  padding-left: 15px;
  padding-right: 15px;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (max-width:76.25em) {
  .emotion-2 {
    margin: 0 10px;
  }
}

.emotion-0 {
  color: #ffffff;
  font-family: Inter,sans-serif;
  font-size: 22px;
  font-weight: 400;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  line-height: 29px;
  margin-bottom: 37px;
}

.emotion-58 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 30px 0 15px;
}

.emotion-8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 150px;
  width: 12.5%;
}

.emotion-8:hover svg > g,
.emotion-8:hover svg + span {
  opacity: 1;
  -webkit-transition: all ease-in 0.3s;
  transition: all ease-in 0.3s;
}

@media (max-width:60em) {
  .emotion-8 {
    width: 33%;
    min-height: 121px;
    margin-bottom: 41px;
  }
}

.emotion-4 {
  max-width: 61px;
  width: 100%;
  margin-bottom: 22px;
}

.emotion-6 {
  color: #ffffff;
  font-family: Inter,sans-serif;
  display: block;
  font-size: 13px;
  font-weight: 400;
  line-height: 17px;
  opacity: 0.7;
  text-align: center;
}

.emotion-62 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-60 {
  color: #ffffff;
  font-size: 14px;
  font-weight: 500;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  margin: 0;
  position: relative;
  cursor: pointer;
}

.emotion-60:before {
  display: block;
  content: '';
  position: absolute;
  width: 0;
  height: 0;
  border: solid 6px transparent;
  border-top-color: #ffffff;
  right: -22px;
  top: 6px;
  border-top-width: 8px;
}

<div>
  <section
    class="emotion-66 emotion-67"
  >
    <div
      class="emotion-64 emotion-65"
    >
      <header
        class="emotion-2 emotion-3"
      >
        <span
          class="emotion-0 emotion-1"
        >
          Browse tickets by Sport
        </span>
      </header>
      <article
        class="emotion-58 emotion-59"
      >
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 0
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 1
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 2
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 3
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 4
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 5
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 6
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 7
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 8
          </span>
        </a>
      </article>
      <div
        aria-label="more sports"
        class="emotion-62 emotion-63"
      >
        <p
          class="emotion-60 emotion-61"
        >
          More
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`<Sports /> Shows more tickets items when more option is clickedRenders sports unchanged 1`] = `
.emotion-120 {
  background: #000000;
  padding: 56px 0 30px 0;
}

@media (max-width:60em) {
  .emotion-120 {
    padding: 24px 0 22px;
  }
}

.emotion-118 {
  max-width: 1255px;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  padding-left: 30px;
  padding-right: 30px;
  padding-left: 15px;
  padding-right: 15px;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (max-width:76.25em) {
  .emotion-2 {
    margin: 0 10px;
  }
}

.emotion-0 {
  color: #ffffff;
  font-family: Inter,sans-serif;
  font-size: 22px;
  font-weight: 400;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  line-height: 29px;
  margin-bottom: 37px;
}

.emotion-112 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 30px 0 15px;
}

.emotion-8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 150px;
  width: 12.5%;
}

.emotion-8:hover svg > g,
.emotion-8:hover svg + span {
  opacity: 1;
  -webkit-transition: all ease-in 0.3s;
  transition: all ease-in 0.3s;
}

@media (max-width:60em) {
  .emotion-8 {
    width: 33%;
    min-height: 121px;
    margin-bottom: 41px;
  }
}

.emotion-4 {
  max-width: 61px;
  width: 100%;
  margin-bottom: 22px;
}

.emotion-6 {
  color: #ffffff;
  font-family: Inter,sans-serif;
  display: block;
  font-size: 13px;
  font-weight: 400;
  line-height: 17px;
  opacity: 0.7;
  text-align: center;
}

.emotion-116 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-114 {
  color: #ffffff;
  font-size: 14px;
  font-weight: 500;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  margin: 0;
  position: relative;
  cursor: pointer;
}

.emotion-114:before {
  display: block;
  content: '';
  position: absolute;
  width: 0;
  height: 0;
  border: solid 6px transparent;
  border-top-color: #ffffff;
  right: -22px;
  top: 6px;
  border-top-width: 8px;
}

<div>
  <section
    class="emotion-120 emotion-121"
  >
    <div
      class="emotion-118 emotion-119"
    >
      <header
        class="emotion-2 emotion-3"
      >
        <span
          class="emotion-0 emotion-1"
        >
          Browse tickets by Sport
        </span>
      </header>
      <article
        class="emotion-112 emotion-113"
      >
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 0
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 1
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 2
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 3
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 4
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 5
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 6
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 7
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 8
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 9
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 10
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 11
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 12
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 13
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 14
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 15
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 16
          </span>
        </a>
        <a
          aria-label="sport ticket"
          class="emotion-8 emotion-9"
          href="/"
        >
          <img
            class="emotion-4 emotion-5"
            src="Sport club"
          />
          <span
            class="emotion-6 emotion-7"
          >
            Sport 17
          </span>
        </a>
      </article>
      <div
        aria-label="more sports"
        class="emotion-116 emotion-117"
      >
        <p
          class="emotion-114 emotion-115"
        >
          More
        </p>
      </div>
    </div>
  </section>
</div>
`;
